src/lib/acks.ts

```diff
@@ -11,61 +11,112 @@ import {
   AckTransactionSet,
   UsageIndicatorCodeSchema,
 } from "./types/PartnerRouting.js";
+import { format } from "date-fns";
+import { ErrorWithContext } from "./errorWithContext.js";
 
 export interface AckDeliveryInput {
   ackTransactionSet: AckTransactionSet;
   interchange: x12.Interchange;
-  edi: string;
+  functionalGroup: x12.FunctionalGroup;
   sendingPartnerId: string;
   receivingPartnerId: string;
 }
 
 export const deliverAck = async (
   input: AckDeliveryInput
 ): Promise<DeliveryResult[]> => {
-  const {
-    ackTransactionSet,
-    interchange,
-    edi,
-    sendingPartnerId,
-    receivingPartnerId,
-  } = input;
+  const { ackTransactionSet, interchange, functionalGroup } = input;
   const usageIndicatorCode = UsageIndicatorCodeSchema.parse(
     interchange.envelope?.usageIndicatorCode
   );
 
+  if (!interchange.envelope || !functionalGroup.envelope) {
+    return [];
```

Comment:
nit: should this be an error? my expectation is that we'd only call `deliverAck` when these envelopes are available and there is something to be ack'd

Reply:
@RossWilliams I definitely agree that the check should happen here. my feedback was more that I think this might be better treated as an error (throw here instead of a silent no-op, returning an empty array)